### PR TITLE
ENH: Window functions should not accept non-integer or negative window lengths

### PR DESCRIPTION
**Pull request by endolith**
Fix for https://github.com/scipy/scipy/issues/6390

I didn't try to consolidate all the identical code because I couldn't see a clean way to do it.  Can always be done later.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
What about:
```
def _check_len(m)
    if ...:
        raise ...
    if m <= 1:
         return np.ones(m)
    else:
        return m


def whatever(...)
     ...
     m = _check_len(m)
     if isinstance(m, ndarray):
         return m
```
That would cut down on the changed lines a lot.

**Comment by endolith** [comment]
Ok I did something a little different, noticing that ones() could handle both cases.

File: scipy/signal/tests/test_windows.py [hunk 1]
```diff
@@ -147,16 +154,20 @@ def test_exponential():
...
     (4, 1.0, True): array([0.0, 0.75, 0.75, 0.0]),
     (4, 0.5, False): array([0.0, 1.0, 1.0, 1.0]),
-    (4, 0.9, False): array([0.0, 0.58682408883346526, 1.0, 0.58682408883346526]),
+    (4, 0.9, False): array([0.0, 0.58682408883346526,
+                            1.0, 0.58682408883346526]),
     (4, 1.0, False): array([0.0, 0.5, 1.0, 0.5]),
     (5, 0.0, True): array([1.0, 1.0, 1.0, 1.0, 1.0]),
-    (5, 0.8, True): array([0.0, 0.69134171618254492, 1.0, 0.69134171618254492, 0.0]),
+    (5, 0.8, True): array([0.0, 0.69134171618254492,
+                           1.0, 0.69134171618254492, 0.0]),
```
Comment (Eric89GXL) [inline scipy/signal/tests/test_windows.py]: These all look like PEP8 changes. Did you add any new tests for the `sym` case changes?

Comment (endolith) [inline scipy/signal/tests/test_windows.py]: I separated them by commit, so test additions are in https://github.com/scipy/scipy/pull/6407/commits/9e1de47fbd2899fa6746bddeb7d2101936ea0a55 and  https://github.com/scipy/scipy/pull/6407/commits/cfe773ad3804d9bb4f5b42d74a74d8db11981126

Comment (Eric89GXL) [inline scipy/signal/tests/test_windows.py]: Ahh sorry I got these changes confused with ones that would fix #2190

**Comment by Eric89GXL** [comment]
LGTM, @e-q do you have time to look and merge if you're happy?

**Comment by e-q** [comment]
This is nice! Thanks @endolith

**Pull request merged**